wip/leetcode-2-add-two-numbers.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * https://leetcode.com/problems/add-two-numbers/
+ */
+
+/**
+ * Definition for singly-linked list.
+ * function ListNode(val) {
+ *     this.val = val;
+ *     this.next = null;
+ * }
+ */
+/**
+ * @param {ListNode} l1
+ * @param {ListNode} l2
+ * @return {ListNode}
+ */
+var addTwoNumbers = function(l1, l2) {
+    const fakeHead = {}
+
+    let current = fakeHead
+    let carry = 0
+
+    while (carry > 0 || l1 || l2) {
+        let val = ((l1 && l1.val) || 0) + ((l2 && l2.val) || 0) + carry
```

Comment:
I believe parentheses around `&&` expression are redundant since it evaluates first.

Reply:
Yeah, may be... Unfortunately, not the most complicated piece of code so I'd leave as is :-)